pym/bob/cmds/build.py

```diff
@@ -458,15 +442,43 @@ def _runShell(self, step, scriptName):
         elif self.__verbose >= 2:
             cmdLine.append('-vv')
 
-        proc = subprocess.Popen(cmdLine, cwd=step.getWorkspacePath(), env=runEnv)
+        oup = pty.openpty()
+        erp = pty.openpty()
+
+        logFile = open(os.path.join(step.getWorkspacePath(), '..', 'log.txt'), 'a')
+        proc = subprocess.Popen(cmdLine, cwd=step.getWorkspacePath(), env=runEnv,
+                universal_newlines=True, stdout=oup[1], stderr=erp[1])
+
+        ansi_escape = re.compile(r'\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]')
+
         try:
-            if proc.wait() != 0:
+            while 1:
+                timeout = .1
+                ready, _, _ = select.select([oup[0], erp[0]], [], [], timeout)
+                if ready:
+                    for r in ready:
+                        data = os.read(r, 1024)
+                        if ((r == erp[0]) and (self.__verbose >= 0)) or ((r == oup[0]) and (self.__verbose >= 1)):
+                            sys.stdout.buffer.write(data)
+                        logFile.write(ansi_escape.sub('', data.decode(sys.stdout.encoding)).replace('\r\n', '\n'))
+                else: # select timeout
+                    if proc.poll() is not None:
+                        break # proc exited
+            if proc.returncode != 0:
+                sys.stdout.flush()
                 raise BuildError("Build script {} returned with {}"
                                     .format(absRunFile, proc.returncode),
                                  help="You may resume at this point with '--resume' after fixing the error.")
         except KeyboardInterrupt:
+            sys.stdout.flush()
```

Comment:
What does happen if you leave that out? Looks to me like this might be a more general fix and should go to scripts.py where the exception is printed.